### ISSUE TITLE
Should we complete windows ohai files?

### ISSUE THREAD

**Issue by Annih** [customink/fauxhai#203]
Hello,

**TL;DR:** Do you think it would be usefull to have more specific windows ohais (i386, x64, CORE)? _- could be extended to other platforms_

I'm currently adding test to one of my cookbooks, and wanted to validate them using most of the windows fauxhai already available. Doing so I encounter a small issue regarding the machine architecture... sometimes I need to test theoric behavior on specific windows "setup" - 32/64bit, CORE.
I also faced the absence of older versions of windows e.g. 7 or Vista/Server 2008.

It's feasible to take one existing fauxhai and transform it to another version by overriding specifics attributes, but it's adding complexity to the tests.

Here is a list of possible fauxhai files - bold for already existing versions:
* Server 2003R2 (**i386**, x64)
* XP (i386, x64)
* Server 2008 (i386, x64, Core i386, Core x64)
* Vista (i386, x64)
* Server 2008R2 (**x64**, Core x64)
* 7 (i386, x64)
* Server 2012 (x64, Core x64)
* 8 (i386, **x64**)
* Server 2012R2 (**x64**, Core x64)
* 8.1 (i386, **x64**)
* 10 (i386, **x64**)

Some of the windows version I listed are old, but I chose to start with 2003R2 generation because fauxhai already support it :) 
Current fauxhai represents 7 / 24 possibilities, but maybe we don't need to support all of them - I already skip the _itanium_ arch :)

If this issue gathers enough interest, I would be able to provide most of the missing fauxhai files; then we should also define a naming convention e.g. _\<**version**>-\<**arch**>[-**CORE**]_

So for people who read everything, what do you think?

**Comment by coderanger** [comment]
We don't support Chef on 32-bit Windows so adding it here seems a bit pointless. Not sure what you mean by "Core x64".

**Comment by Annih** [comment]
Wow thanks @coderanger for looking at this issue, I almost forgot I created it.
Ok for the 32-bit platforms, lets get rid of them.

By `Core x64` I mean Windows Server in its Core version, because Windows Server does not support 32bits since 2008R2 the x64 part is useless :). When I was playing with it (2008R2 Core) I discovered that my code needed to handle it properly, based on some Ohai info.
The issue I faced at this time, was that there were no Fauxhai for this platform, so I created this issue.

Starting with Windows Server 2016, Microsoft introduced the `Nano` version, but I didn't play with it, so I don't know how it integrates with Chef.

With Windows Server 2016 RS3, Microsoft kept the `Nano` &  `Core` versions, but removed the `normal` one, so next fauxhai data for Windows Server 2016 will have to handle the Core version anyway :D

I'm going to update my original issue message to take your feedback into account.

**Comment by coderanger** [comment]
That would all depend on how Ohai deals with Windows Nano and Server Core, I would have assumed that was a new platform, not a version on an existing one. I don't think we have Nano-compatible installers for Chef, so utility there is probably pretty minimal but whatever data Ohai produces we would be happy to add.

**Comment by Annih** [comment]
We could consider `Server Core` as a new version, but how to represent that in Chef? Changing this might have some side effects for existing environments, IMHO this is not worth it.
Currently you can use the [::Chef::ReservedNames::Win32::Version.core? helper](https://github.com/chef/chef/blob/35458abec91b644eb56b5bd38dd413391c069b63/lib/chef/win32/version.rb#L110) to determine in which "version" you are, it working fine - even if I prefer the [::Windows::VersionHelper.core_version? library from the windows cookbook](https://github.com/chef-cookbooks/windows/blob/master/libraries/version_helper.rb#L51).

`Core` servers may have different capabilities and features names but are pretty much the same as normal servers. The only difference I can see for fauxhai is the `kernel.os_info.operating_system_sku` attribute.

**Comment by coderanger** [comment]
That discussion is for Ohai, not here. Whatever Ohai does, Fauxhai will follow.

**Comment by Annih** [comment]
Now that I updated the list of available Windows versions, I think my suggested naming convention is obsolete :smile:
How to expose all the Windows 10 versions?
* Using the code name - TH1, TH2, RS1, RS2, RS3
* Using the date/version - 1507, 1511, 1607, 1703, 1709
* Using the build - 10240, 10586, 14393, 15063, 16299

**Comment by coderanger** [comment]
Again, platform and platform version naming is not up to Fauxhai. We match the output of Ohai for the most part. You will have to take it up over there.

**Comment by Annih** [comment]
You miss understood me, I'm no talking about changing anything in Ohai values.
Currently if I want to get Windows Server 2016 fauxhai data, the `platform` is **windows** and the `platform_version` is **2016**. 

I want to setup a naming convention in Fauxhai to easily select the desired and precise platform data in a coherent way.

For instance I would still use the **windows** platform, but maybe I should use **Server-2016-RS1** as desired `platform_version`, or **Server_2016_1607**.
Or maybe I should use a different `platform` **windows_server** and precise **1607** as `platform_version`

Are my intentions clearer?

**Comment by coderanger** [comment]
We don't really have the resources to commit to maintaining more dumps than we already have. If you need more specific data, it's probably something that should be in your unit tests themselves.

**Comment by Annih** [comment]
Ok that was my original question. Is it interesting for Chef and the community that we provide all dumps?
Windows has currently only 9 dumps over the 19 supported versions. CentOS has more than 30 dumps :P

If some of the missing were provided, would they be accepted?

**Comment by tas50** [comment]
@Annih We could certainly dump more versions. We have a ton of Linux dumps because for a week I installed every single one of those and ran fauxhai on it. It took a REALLY long time, but it's orders of magnitude faster than installing Windows. I dumped several versions of Windows, but it's certainly not all there.  I'll take a look at what Chef / Microsoft supports and see if there's anywhere we could fill in some gaps. We won't be dumping Windows 2008, or Windows 8 though since those aren't supported platforms. We should make sure we have the latest Windows 10 though and we should probably version it on the build numbers since MS has gone to a bit of a rolling release model now.
